### PR TITLE
Only allow state filter for filter card

### PR DESCRIPTION
**Pull request by balloob**
Rework filter card to require hardcoded entities and only allow filtering by state.

```yaml
views:
  - title: Filter
    cards:
      - type: entity-filter
        entities:
         - light.bed_light
         - light.kitchen_lights
         - light.ceiling_lights
        state_filter:
         - "on"
        card:
          title: Filter for on
      - type: entity-filter
        entities:
         - light.bed_light
         - light.kitchen_lights
         - light.ceiling_lights
        state_filter:
         - "on"
        card:
          title: Filter for on + Glance
```

Also includes a cleanup so that we validate and calculate entities config only once.

### CODE REVIEW

**Comment by ciotlosm** [comment]
I assume this makes https://github.com/home-assistant/ui-schema/issues/47 and https://github.com/home-assistant/ui-schema/issues/41 obsolete

**Comment by balloob** [comment]
Yes

**Comment by ciotlosm** [comment]
@balloob will there still be a way to show all entities in a card? Would be nice for debugging what you have missing.

File: src/panels/lovelace/cards/hui-entities-card.js [hunk 1]
```diff
@@ -5,8 +5,7 @@ import { PolymerElement } from '@polymer/polymer/polymer-element.js';
 import stateCardType from '../../../common/entity/state_card_type.js';
 import computeDomain from '../../../common/entity/compute_domain.js';
 import { DOMAINS_HIDE_MORE_INFO } from '../../../common/const.js';
-import computeConfigEntities from '../common/compute-config-entities';
-import validateEntitiesConfig from '../common/validate-entities-config';
+import processEntitiesConfig from '../common/process-config-entities.js';
```
Comment (jeradM) [inline src/panels/lovelace/cards/hui-entities-card.js]: `import processConfigEntities` for consistency

**Comment by balloob** [comment]
We can add that in a different card.

**Comment by danrspencer** [comment]
What was the reason behind removing this? Not having to create a hardcoded list of entities in a group was/is an extremely useful feature.

**Comment by balloob** [comment]
It was making things overly complicated. One can make a custom card if you want this flexibility. PRs are not the place to discuss this though.

**Pull request merged**